### PR TITLE
Logic functionality is added

### PR DESCRIPTION
**Pull request by larsbjn**
* Logic functionality is added

### CODE REVIEW

File: SensumUdred_Business/src/business/logic/Case.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    private String state;
+    private Date openingDate;
+    private Date closingDate;
+    private int id;
+    private boolean consent;
+    private String reason;
+    private String availableOffers;
+    private String sourceOfReqeust;
+    private Citizen citizen;
+    private CaseWorker caseWorker;
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Case.java]: Javadoc

File: SensumUdred_Business/src/business/common/ILogicFacade.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ */
+public interface ILogicFacade {
+    ICase openCase(ICitizenData citizenData);
+    boolean closeCase(int caseId);
+    ICase findActiveCase(int value, boolean isCpr);
+    ICase findActiveCase(String name);
+    List<? extends ICase> getAllActiveCases();
+    List<? extends ICase> getActiveCases();
+    void setCaseWorker(String userId);
+    void removeCaseWorker();
```
Comment (Frede175) [inline SensumUdred_Business/src/business/common/ILogicFacade.java]: Javadoc

File: SensumUdred_Business/src/business/logic/Case.java [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+        this.citizen = citizen;
+        this.caseWorker = caseWorker;
+    }
+
+    /**
+     * Construct an case with already exisiting case data
+     * @param c The case data to extract from
+     */
+    public Case(ICase c) {
+
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Case.java]: Missing impl for loading

File: SensumUdred_Business/src/business/logic/Case.java [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    public String getCitizenName() {
+        return citizen.getName();
+    }
+
+    /**
+     * Closes this case
+     * @return True if the case is closed correct
+     */
+    public boolean closeCase() {
+        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Case.java]: Missing

File: SensumUdred_Business/src/business/logic/CaseWorker.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+import common.ICase;
+import common.ICaseWorker;
+import common.ICitizen;
+import common.ICitizenData;
+import java.util.List;
+
+/**
+ *
+ * @author andreasmolgaard-andersen
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/CaseWorker.java]: Author

File: SensumUdred_Business/src/business/logic/CaseWorker.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        this.department = department;
+    }
+
+    /**
+     * Opens a new case to the case worker
+     * @param data The data to put into the new case
+     * @return The new case opened
+     */
+    public Case openCase(ICitizenData data) {
+        List<? extends ICitizen> citizens = department.getCitizens();
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/CaseWorker.java]: This should just be List<ICitizen>

File: SensumUdred_Business/src/business/logic/CaseWorker.java [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        c.setConsent(data.getConsent());
+        c.setReason(data.getReason());
+        c.setSourceOfReqeust(data.getSourceOfRequest());
+        c.setState(data.getState());
+        return true;
+    }
+    
+    /**
+     * Saves the 
+     * @return
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/CaseWorker.java]: Javadoc

File: SensumUdred_Business/src/business/logic/CaseWorker.java [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        c.setState(data.getState());
+        return true;
+    }
+    
+    /**
+     * Saves the 
+     * @return 
+     */
+    public boolean saveCase() {
+        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/CaseWorker.java]: This should not be here

File: SensumUdred_Business/src/business/logic/Citizen.java [hunk 9]
```diff
@@ -0,0 +1,107 @@
+package business.logic;
+
+import common.Gender;
+import common.ICitizen;
+import common.RelationshipStatus;
+
+/**
+ *
+ * @author andreasmolgaard-andersen
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Citizen.java]: Author

File: SensumUdred_Business/src/business/logic/Citizen.java [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+    @Override
+    public int getCpr() {
+        return cpr;
+    }
+    
+    /**
+     * Remove the citizens active case
+     */
+    public void removeActiveCase() {
+
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Citizen.java]: Missing

File: SensumUdred_Business/src/business/logic/Department.java [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+public class Department implements IDepartment {
+
+    private List<CaseWorker> caseWorkers;
+    private List<Case> inactiveCases;
+    private List<Citizen> citizens;
+    private String name;
+    private String treatmentArea;
+    private String address;
+    private String email;
+    private String phoneNumber;
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Department.java]: Javadoc

File: SensumUdred_Business/src/business/logic/Department.java [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+        this.phoneNumber = phoneNumber;
+    }
+    
+    /**
+     * Gets the name of the department
+     * @return The name of the department
+     */
+    @Override
+    public String getName() {
+        return this.name;
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Department.java]: No need for this here. There is a lot of 'this' in this class

File: SensumUdred_Business/src/business/logic/Department.java [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+    @Override
+    public List<? extends ICase> getAllActiveCases() {
+        List<ICase> cases = new ArrayList();
+        for (CaseWorker caseWorker : caseWorkers) {
+            cases.addAll(caseWorker.getActiveCases());
+        }
+        return cases;
+    }
+    
+    public ICase findActiveCase(int value, boolean isCpr) {
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Department.java]: Javadoc

File: SensumUdred_Business/src/business/logic/Department.java [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+            if(isCpr) {
+                if(value == activeCase.getCitizen().getCpr()) return activeCase;
+            } else {
+                if(value == activeCase.getId()) return activeCase;
+            }
+        }
+        return null;
+    }
+    
+    public ICase findActiveCase(String name) {
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Department.java]: Javadoc

File: SensumUdred_Business/src/business/logic/LogicFacade.java [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+import business.common.ILogicFacade;
+import common.ICase;
+import common.ICaseWorker;
+import common.ICitizenData;
+import common.IDepartment;
+import java.util.List;
+
+/**
+ *
+ * @author larsjorgensen
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/LogicFacade.java]: Author

File: SensumUdred_Business/src/business/logic/Person.java [hunk 16]
```diff
@@ -0,0 +1,36 @@
+package business.logic;
+
+import common.IPerson;
+
+/**
+ *
+ * @author andreasmolgaard-andersen
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Person.java]: Author

File: SensumUdred_Business/src/business/logic/Person.java [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+ */
+public class Person implements IPerson{
+
+    private String name;
+    private String phoneNumber;
+    private String email;
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Person.java]: Javadoc

File: SensumUdred_Business/src/business/logic/Person.java [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+
+    public Person(String name, String phoneNumber, String email) {
+        this.name = name;
+        this.phoneNumber = phoneNumber;
+        this.email = email;
+    }
+    
+    @Override
+    public String getName() {
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Person.java]: Javadoc

File: SensumUdred_Business/src/business/logic/Person.java [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+        return this.name;
+    }
+
+    @Override
+    public String getPhoneNumber() {
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Person.java]: Javadoc

File: SensumUdred_Business/src/business/logic/Person.java [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+        return this.phoneNumber;
+    }
+
+    @Override
+    public String getEmail() {
```
Comment (Frede175) [inline SensumUdred_Business/src/business/logic/Person.java]: Javadoc

File: SensumUdred_Common/src/common/IPerson.java [hunk 21]
```diff
@@ -11,6 +11,6 @@
  */
 public interface IPerson {
     String getName();
-    String getPhoneNumber();
+    String  getPhoneNumber();
```
Comment (Frede175) [inline SensumUdred_Common/src/common/IPerson.java]: See if you can change this back!

File: SensumUdred_Business/src/business/common/ILogicFacade.java [hunk 22]
```diff
@@ -0,0 +1,70 @@
...
+
+import common.ICase;
+import common.ICitizenData;
+import java.util.List;
+
+
+
+/**
+ *
+ * @author larsjorgensen
```
Comment (muggell) [inline SensumUdred_Business/src/business/common/ILogicFacade.java]: Author needs all members of the team

File: SensumUdred_Business/src/business/common/ILogicFacade.java [hunk 23]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Lars Bjerregaard Jørgensen (larjo17)
+ * Software Engineering at SDU
```
Comment (muggell) [inline SensumUdred_Business/src/business/common/ILogicFacade.java]: Header needs to be removed

File: SensumUdred_Business/src/business/logic/Case.java [hunk 24]
```diff
@@ -0,0 +1,264 @@
+package business.logic;
+
+import common.ICase;
+import common.ICaseWorker;
+import common.ICitizen;
+import java.util.Date;
+
+/**
+ *
+ * @author andreasmolgaard-andersen
```
Comment (muggell) [inline SensumUdred_Business/src/business/logic/Case.java]: Author: All members of the team

File: SensumUdred_Business/src/business/logic/LogicFacade.java [hunk 25]
```diff
@@ -0,0 +1,124 @@
+/*
+ * Lars Bjerregaard Jørgensen (larjo17)
+ * Software Engineering at SDU
```
Comment (muggell) [inline SensumUdred_Business/src/business/logic/LogicFacade.java]: Header needs to be removed

File: SensumUdred_Business/src/business/logic/Case.java [hunk 26]
```diff
@@ -0,0 +1,264 @@
...
+     * Gets the case worker of the case
+     * @return The case worker of the case
+     */
+    @Override
+    public ICaseWorker getCaseWorker() {
+        return caseWorker;
+    }
+
+    /**
+     * Sets so the citizen is informed
```
Comment (krongrah) [inline SensumUdred_Business/src/business/logic/Case.java]: *Sets the citizen's status/state to informed.

File: SensumUdred_Business/src/business/logic/Case.java [hunk 27]
```diff
@@ -0,0 +1,264 @@
...
+     */
+    public void setAvailableOffers(String availableOffers) {
+        this.availableOffers = availableOffers;
+    }
+
+    /**
+     * Sets the source of request
+     * @param sourceOfReqeust The source of request
+     */
+    public void setSourceOfReqeust(String sourceOfReqeust) {
```
Comment (krongrah) [inline SensumUdred_Business/src/business/logic/Case.java]: guess what

File: SensumUdred_Business/src/business/logic/CaseWorker.java [hunk 28]
```diff
@@ -0,0 +1,146 @@
+package business.logic;
+
+import common.ICase;
+import common.ICaseWorker;
+import common.ICitizen;
+import common.ICitizenData;
+import java.util.List;
+
+/**
+ * @author Andreas Mølgaard-Andersen
```
Comment (krongrah) [inline SensumUdred_Business/src/business/logic/CaseWorker.java]: class misses javadoc

File: SensumUdred_Business/src/business/logic/Citizen.java [hunk 29]
```diff
@@ -0,0 +1,134 @@
+package business.logic;
+
+import common.Gender;
+import common.ICase;
+import common.ICitizen;
+import common.RelationshipStatus;
+
+/**
```
Comment (krongrah) [inline SensumUdred_Business/src/business/logic/Citizen.java]: add javadoc

**Pull request merged**